### PR TITLE
Fixed issue 2945

### PR DESCRIPTION
**Pull request by akete**

### CODE REVIEW

**Comment by obiltschnig** [comment]
I'm still seeing test suite failures on Travis.
https://travis-ci.org/pocoproject/poco/builds/656324730?utm_source=slack&utm_medium=notification

File: Foundation/testsuite/src/VarTest.cpp [hunk 1]
```diff
@@ -2974,7 +2974,17 @@ void VarTest::testIterator()
 {
 	Var da;
 	assertTrue (da.isEmpty());
-	assertTrue (da.begin() == da.end());
+	bool thrown = false;
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp]: not needed

File: Foundation/testsuite/src/VarTest.cpp [hunk 2]
```diff
@@ -2974,7 +2974,17 @@ void VarTest::testIterator()
...
+	try {
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp]: style: opening brace on new line

File: Foundation/testsuite/src/VarTest.cpp [hunk 3]
```diff
@@ -2974,7 +2974,17 @@ void VarTest::testIterator()
...
+		auto it = da.begin();
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp]: add `fail("calling begin() on empty Var must throw");` here

File: Foundation/testsuite/src/VarTest.cpp [hunk 4]
```diff
@@ -2974,7 +2974,17 @@ void VarTest::testIterator()
...
+	}
+	catch (const InvalidAccessException&) {
+		thrown = true;
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp]: not needed

File: Foundation/testsuite/src/VarTest.cpp [hunk 5]
```diff
@@ -2974,7 +2974,17 @@ void VarTest::testIterator()
...
+	}
+	assertTrue(thrown);
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp]: not needed

File: Foundation/testsuite/src/VarTest.cpp [hunk 6]
```diff
@@ -2975,13 +2975,12 @@ void VarTest::testIterator()
 	Var da;
 	assertTrue (da.isEmpty());
 	bool thrown = false;
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp]: still not needed

Comment (akete) [reply Foundation/testsuite/src/VarTest.cpp]: :man_facepalming: duh...

**Pull request merged**